src/index.js

```diff
@@ -0,0 +1,88 @@
+const crypto = require('crypto');
+const assert = require('assert');
+
+
+/**
+ * Enum for two state values.
+ * @readonly
+ * @enum {number}
+ */
+const actionEnum = {
+  ENCRYPT: 0,
+  DECRYPT: 1
+};
+
+/**
+   * Encrypt/Decrypt incoming data depending on action
+   * @param  {Array|string|boolean|object} data - data to encrypt/decrypt.
+   * In case of encryption, data can be Array of string or string.
+   * In case of decryption, data can be an Array of objects or an object
+   * @param {enum} action - action to perform. Can be either ENCRYPT or DECRYPT
+   * @return {Array|string|boolean|object} - encrypted Array of objects or one object / decrypted Array of strings or one string
+*/
+function performAction(data, key, action) {
+  if ([null, undefined].includes(data) || [null, undefined].includes(key)) {
+    return data;
+  }
+
+  const isArray = Array.isArray(data);
+  const queue = isArray ? [...data] : [data];
+  const result = [];
+
+  for (let item of queue) {
+    if (action === actionEnum.ENCRYPT) {
+      // if already encrypted
+      if (item.content && item.vector && item.tag) {
+        result.push(item);
+        continue; // eslint-disable-line no-continue
+      }
+
+      const itemValue = typeof item === 'boolean' ? `${item}` : item; // converting boolean to string
+      const vector = crypto.randomBytes(12);
+      const cipher = crypto.createCipheriv('aes-256-gcm', key, vector);
+      const encryptedItem = `${cipher.update(itemValue, 'utf8', 'hex')}${cipher.final('hex')}`;
+      const tag = cipher.getAuthTag();
+      result.push({
+        content: encryptedItem,
+        vector: vector.toString('hex'),
+        tag: tag.toString('hex')
+      });
+    } else if (action === actionEnum.DECRYPT) {
+      // if was not encrypted
+      if (!item.content || !item.vector || !item.tag) {
+        result.push(item);
+        continue; // eslint-disable-line no-continue
+      }
+
+      const decipher = crypto.createDecipheriv('aes-256-gcm', key, Buffer.from(item.vector, 'hex'));
+      decipher.setAuthTag(Buffer.from(item.tag, 'hex'));
+      const decryptedValue = `${decipher.update(item.content, 'hex', 'utf8')}${decipher.final('utf8')}`;
+      // converting string to boolean
+      const normalizedValue = ['true', 'false'].includes(decryptedValue) ? decryptedValue === 'true' : decryptedValue;
+      result.push(normalizedValue);
+    }
+  }
+  return isArray ? result : result[0];
+}
+
```

Comment:
### Suggestion:
Maybe refactor the `performAction` function into two functions: `encrypt` and `decrypt` this in order to avoid the `if`, `else if` check for each loop on the data array.
For now i see no benefits of the `performAction` function.

Reply:
or you can create an object with two function `encrypt` and `decrypt` and then on the for loop you can do 
```js
for (let item of queue) {
  result.push(object[action](data, key));
}
``` 
But you need to modify your enum, so the value should represent the function name as:
```js
const actionEnum = {
   ENCRYPT: 'encrypt',
   DECRYPT: 'decrypt'
};
```

Reply:
Yeah writing 2 separate functions seems good.
The user wouldn't need the to precise action, just call the right function with the data and the key